### PR TITLE
Bug 1305403 -- env vars to override route hostname

### PR DESCRIPTION
**Pull request by tnguyen-rh**
@ramr 
This is the first part of bug 1305403 (the second part, router sharding proper, will be in another PR).  WDYT?

### CODE REVIEW

File: install_config/install/deploy_router.adoc [hunk 1]
```diff
@@ -234,39 +236,71 @@ router] on your OpenShift cluster using IP failover.
...
++
+====
+The following example modifies the deployment configuration named `router` to
+use a custom template, and arranges to propagate the change through the cluster.
+----
+$ oc env dc/router \
+    ROUTER_OVERRIDE_HOSTNAME=true \
+    ROUTER_SUBDOMAIN='${name}-${namespace}.apps.example.com'
+
+$ oc scale dc/router --replicas=<n>
```
Comment (ramr) [inline install_config/install/deploy_router.adoc]: You don't need the `oc scale` in the example here. `oc env dc/router ...` would create a newer version and deploy it.

File: install_config/install/deploy_router.adoc [hunk 2]
```diff
@@ -234,39 +236,71 @@ router] on your OpenShift cluster using IP failover.
...
+Their influence depends on whether or not a route has a hostname.
+
+`ROUTER_OVERRIDE_HOSTNAME`:::
+This variable comes into play when a route *has* a hostname.
+If the value is `true`, OpenShift ignores that hostname.
+If the value is `false`, OpenShift accepts that hostname.
+
+`ROUTER_SUBDOMAIN`:::
+This variable comes into play when a route *has no* hostname.
+The value specifies a template to generate the hostname.
```
Comment (ramr) [inline install_config/install/deploy_router.adoc]: Maybe say:  `For routes that do not have a hostname, the router will use the router subdomain template to automatically generate a name for the route upon its admission.`  or something to that effect.

File: install_config/install/deploy_router.adoc [hunk 3]
```diff
@@ -234,39 +236,71 @@ router] on your OpenShift cluster using IP failover.
...
+
+You can use the `oc env` command to set environment variables
+on the router's deployment configuration object.
+Two variables control if and how the subdomain is formed.
+Their influence depends on whether or not a route has a hostname.
+
+`ROUTER_OVERRIDE_HOSTNAME`:::
+This variable comes into play when a route *has* a hostname.
+If the value is `true`, OpenShift ignores that hostname.
+If the value is `false`, OpenShift accepts that hostname.
```
Comment (ramr) [inline install_config/install/deploy_router.adoc]: Maybe something like `This flag controls whether a hostname specified in a route [spec.host] is accepted/admitted/allowed by the router. If this is set/true, then the hostname in a route is always overridden by with the hostname template (specified via the ROUTER_SUBDOMAIN env variable).`

File: install_config/install/deploy_router.adoc [hunk 4]
```diff
@@ -234,39 +236,71 @@ router] on your OpenShift cluster using IP failover.
...
 myroute-mynamespace.v3.openshift.test
 ----
 ====
 
+Setting environment variables in the deployment configuration::
+
+You can use the `oc env` command to set environment variables
+on the router's deployment configuration object.
+Two variables control if and how the subdomain is formed.
+Their influence depends on whether or not a route has a hostname.
```
Comment (ramr) [inline install_config/install/deploy_router.adoc]: Hmm, subdomain formed sounds a bit weird ... maybe something akin to : `You can control the hostnames [route spec.host] fields that the router will allow/admit using the following environment variables: `  and then describe the 2 variables below.

File: install_config/install/deploy_router.adoc (hunk 1 above)
Comment (tnguyen-rh) [inline install_config/install/deploy_router.adoc]: @ramr Thanks.  I've removed the `oc scale` command.

File: install_config/install/deploy_router.adoc (hunk 2 above)
Comment (tnguyen-rh) [inline install_config/install/deploy_router.adoc]: @ramr I've made a similar change, using the term "admission".  I wonder, however, if that term needs to be defined more formally, or if perhaps, there is a better term.

**Comment by tnguyen-rh** [comment]
@ramr 
Thanks for the suggestions.  I've incorporated their essence (if not their text, exactly) into a new batch of changes.  PTAL.

@adellape @ahardin-rh @tpoitras @bfallonf 
WDYT?

**Comment by ramr** [comment]
LGTM

**Comment by ramr** [comment]
LGTM - for some reason github is slow and previous comment didn't take.

**Comment by tnguyen-rh** [comment]
@adellape @ahardin-rh @tpoitras @bfallonf
WDYT?

**Comment by ahardin-rh** [comment]
LGTM  :ok_hand:

**Comment by adellape** [comment]
[rev_history]
|link:../install_config/install/deploy_router.html[Installing -> Deploying a Router]
|Updated the link:../install_config/install/deploy_router.html#customizing-the-default-routing-subdomain[Customizing the Default Routing Subdomain] section for setting environment variables in the deployment configuration.
%

**Pull request merged**